packages/round-manager/src/features/round/ApplicationsApproved.tsx

```diff
@@ -34,7 +34,7 @@ export default function ApplicationsApproved() {
   const { id } = useParams();
   const { chain } = useWallet();
 
-  const { applications, isLoading } = useApplicationByRoundId(id!);
+  const { applications, isLoading } = useApplicationByRoundId(id ?? '');
```

Comment:
Hmm curious if we it's smarter to update the useApplicationByRoundId to accept a nullable field / we update this to make sure this is not invoked if id is missing

Reply:
Updating useApplicationByRoundId to accept nullable field is causing other issues, as the id, is being passed as a parameter to another function within useApplicationByRoundId & the function accepts only a valid string ID.

I feel updating it to make sure it is not invoked isn't needed, as in the next step, only valid approved applications are being filtered out. 

Code link: https://github.com/gitcoinco/grants-round/blob/0e5bb58941789bdc04bb87e0346a5ea4f8ead9e2/packages/round-manager/src/features/round/ApplicationsApproved.tsx#L38

Reply:
Generally you are not supposed to call React hooks conditionally -- you will get an ESLint warning
`React Hook "useApplicationByRoundId" is called conditionally. React Hooks must be called in the exact same order in every component render. (react-hooks/rules-of-hooks)`

Either we can change `useApplicationByRoundId` to accept optional `id`, and return some default values if `id` is not defined. Or disable the non-null assertion rule for this line, with the explanation that the `id` URL parameter should never be null

Reply:
> we disable the non-null assertion rule for this line, with the explanation that the id URL parameter should never be null

I would lean towards this